cmake/Findmlir.cmake

```diff
@@ -0,0 +1,24 @@
+# This file allows users to call find_package(MLIR) and pick up our targets.
+
+set(HAS_MLIR_MIOPEN_SUPPORT False)
+
+if(DEFINED ENV{MLIR_SOURCE_PATH})
+  set(MLIR_SOURCE_PATH   "$ENV{MLIR_SOURCE_PATH}")
+  set(MLIR_BUILD_PATH    "$ENV{MLIR_BUILD_PATH}" )
```

Comment:
We shouldn't rely on env variables to find dependencies. This should just use cmake's find_* command to find the different pieces needed. Ideally, it would be even nicer if mlir could provide a mlir-config.cmake so we dont need to this find module, but this is fine for a prototyping phase.